src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -0,0 +1,3092 @@
+
+
+        internal int _regexopcode;       // the current opcode being processed
+        internal int _codepos;           // the current code being translated
+        internal int _backpos;           // the current backtrack-note being translated
+
+        internal RegexOptions _options;           // options
+
+        // special code fragments
+        internal int[] _uniquenote;        // _notes indices for code that should be emitted <= once
+        internal int[] _goto;              // indices for forward-jumps-through-switch (for allocations)
+
+        // indices for unique code fragments
+        internal const int Stackpop = 0;    // pop one
+        internal const int Stackpop2 = 1;    // pop two
+        internal const int Stackpop3 = 2;    // pop three
+        internal const int Capback = 3;    // uncapture
+        internal const int Capback2 = 4;    // uncapture 2
+        internal const int Branchmarkback2 = 5;    // back2 part of branchmark
+        internal const int Lazybranchmarkback2 = 6;    // back2 part of lazybranchmark
+        internal const int Branchcountback2 = 7;    // back2 part of branchcount
+        internal const int Lazybranchcountback2 = 8;    // back2 part of lazybranchcount
+        internal const int Forejumpback = 9;    // back part of forejump
+        internal const int Uniquecount = 10;
+
+        static RegexCompiler()
+        {
+            // fields
+            s_textbegF = RegexRunnerField("runtextbeg");
+            s_textendF = RegexRunnerField("runtextend");
+            s_textstartF = RegexRunnerField("runtextstart");
+            s_textposF = RegexRunnerField("runtextpos");
+            s_textF = RegexRunnerField("runtext");
+            s_trackposF = RegexRunnerField("runtrackpos");
+            s_trackF = RegexRunnerField("runtrack");
+            s_stackposF = RegexRunnerField("runstackpos");
+            s_stackF = RegexRunnerField("runstack");
+            s_trackcountF = RegexRunnerField("runtrackcount");
+
+            // methods
+            s_ensurestorageM = RegexRunnerMethod("EnsureStorage");
+            s_captureM = RegexRunnerMethod("Capture");
+            s_transferM = RegexRunnerMethod("TransferCapture");
+            s_uncaptureM = RegexRunnerMethod("Uncapture");
+            s_ismatchedM = RegexRunnerMethod("IsMatched");
+            s_matchlengthM = RegexRunnerMethod("MatchLength");
+            s_matchindexM = RegexRunnerMethod("MatchIndex");
+            s_isboundaryM = RegexRunnerMethod("IsBoundary");
+            s_charInSetM = RegexRunnerMethod("CharInClass");
+            s_isECMABoundaryM = RegexRunnerMethod("IsECMABoundary");
+            s_crawlposM = RegexRunnerMethod("Crawlpos");
+            s_checkTimeoutM = RegexRunnerMethod("CheckTimeout");
+
+            s_chartolowerM = typeof(char).GetMethod("ToLower", new Type[] { typeof(char), typeof(CultureInfo) });
+            s_getcharM = typeof(string).GetMethod("get_Chars", new Type[] { typeof(int) });
```

Comment:
What is this referencing?  I don't see any Chars property or get_Chars method on string.

Reply:
Oh, nevermind, it's the indexer.